File: leetcode2/1easy/김대원/Q1154.java
// 1. 아이디어 : 윤년 판단
// 2. 시간복잡도 : O(N)
// 3. 자료구조/알고리즘 : 구현

class Solution {
    public int dayOfYear(String date) {
        int[] days = new int[] {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        int[] info = Arrays.stream(date.split("-")).mapToInt(Integer::parseInt).toArray();
        int year = info[0];
        int month = info[1];
        int day = info[2];

        if (month >= 2 && ((year % 4 == 0 && year % 100 != 0) || year % 400 == 0)) {
            days[1] = 29;
        }

        int answer = day;
        for (int i = 0; i < month - 1; i++) answer += days[i];
        return answer;
    }
}

File: leetcode2/1easy/강병호/Q1154.py
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def dayOfYear(self, date: str) -> int:
        result = 0
        year, month, day = map(int, date.split("-"))

        for i in range(1, month):
            result += self.checkDays(year, i)
        result += day

        return result

    def checkDays(self, year:int, month: int) -> int:
        days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

        if month == 2 and year == 1900:
            return days[month-1]
        if month == 2 and self.isLeapYear(year):
            return 29
        else:
            return days[month-1]

    def isLeapYear(self, year) -> bool:
        if year % 4 == 0:
            return True
        else:
            return False
